File: src/NBdbm/Classes/Interfaces/iEVT.vb
Namespace Interfaces

  Namespace iEVT

    Public Interface iCadastroFuncionarios
      Inherits NBdbm.Interfaces.iCTR.iCadastroUsuario
    End Interface

    Public Interface iCadastroLocalidades
      'Inherits NBdbm.Interfaces.iCTR.iCadastroEntidade
      Sub Salvar(ByVal NoCommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal NoCommit As Boolean)
      Sub Excluir()
      Property Localidade() As Interfaces.iEVT.Primitivas.iLocalidades
    End Interface

    Public Interface iCadastroItens
      Inherits System.IDisposable
      Sub Salvar(ByVal NoCommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal NoCommit As Boolean)
      Sub Excluir()
      Property Item() As Interfaces.iEVT.Primitivas.iItem
      Property xmPath_LinkItemNo() As String
    End Interface

    Public Interface iCadatroKit
      'Inherits Interfaces.Primitivas.allClass - > Não herda, porque ela só aglutina!
      Inherits System.IDisposable
      Sub Salvar(ByVal NoCommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal NoCommit As Boolean)
      Sub Excluir()
      Property Kit() As Interfaces.iEVT.Primitivas.iKit
      'Property xmPath_LinkEntNo() As String 'Se houver nó para os kits
      Property KitItem() As Interfaces.ievt.Primitivas.iKitItem
    End Interface

    Public Interface iCadatroOrcamento
      'Inherits Interfaces.Primitivas.allClass - > Não herda, porque ela só aglutina!
      Inherits System.IDisposable
      Sub Salvar(ByVal NoCommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal NoCommit As Boolean)
      Sub Excluir()
      Property Orc() As Interfaces.iEVT.Primitivas.iOrcamento
      Property OrcItem() As Interfaces.iEVT.Primitivas.iOrcamentoObjs
      Property OrcKit() As Interfaces.iEVT.Primitivas.iOrcamentoObjs
      Property OrcDespAdic() As Interfaces.iEVT.Primitivas.iOrcamentoDespAdicional
      Property OrcParcela() As Interfaces.iEVT.Primitivas.iOrcamentoParcelas
    End Interface

    Public Interface iCadastroCustomROrc
      Inherits System.IDisposable
      Sub Salvar(ByVal nocommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal nocommit As Boolean)
      Sub Excluir()
      Property CustomROrc() As Interfaces.iEVT.Primitivas.iCustomROrc
      Property CustomROrc_Grupos() As Interfaces.iEVT.Primitivas.iCustomROrc_Grupos
      Property CustomROrc_ItemGrupo() As Interfaces.iEVT.Primitivas.iCustomROrc_ItensGrupo
      ReadOnly Property ColCustomROrc_Grupos() As System.Collections.Hashtable
      ReadOnly Property ColCustomROrc_ItensGrupo() As System.Collections.Hashtable
    End Interface

    Public Interface iCadatroOrdemServico
      Inherits System.IDisposable
      Sub Salvar(ByVal NoCommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal NoCommit As Boolean)
      Sub Excluir()
      Property OS() As Interfaces.iEVT.Primitivas.iOrdemServico
      Property OSObjs() As Interfaces.iEVT.Primitivas.iOrdemServicoObjs
    End Interface

    Public Interface iCadastroContasPagarReceber
      Inherits IDisposable
      Sub Salvar(ByVal NoCommit As Boolean)
      Sub Salvar()
      Sub Excluir(ByVal NoCommit As Boolean)
      Sub Excluir()
      Property Conta() As Interfaces.iEVT.Primitivas.iContasPagarReceber
      Property Parcela() As Interfaces.iEVT.Primitivas.iContasPagarReceber_Parcelas
      ReadOnly Property Parcelas() As NBdbm.Fachadas.NbCollection
    End Interface


    Namespace Primitivas

      Public Interface iQualidade
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property qualidade_key() As String
      End Interface

      Public Interface iSituacao
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property situacao_key() As String
      End Interface

      Public Interface iLocalidades
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property idEntidade() As Integer
        Property Nome_key() As String
        Property comoChegar() As String
      End Interface

      Public Interface iDeposito
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property Nome_Key() As String
        Property Endereco() As String
      End Interface

      Public Interface iItem
        Inherits Interfaces.Primitivas.iObjetoTabela

        ReadOnly Property IdObj() As String
        Property Descricao_key() As String
        Property Quantidade() As Integer
        Property ValorLocacao() As Double
        Property DescMax() As Decimal
        Property VidaUtil() As Integer
        Property NumeroSerie() As String
        Property IdQualidade() As Integer
        Property IdDeposito() As Integer
        Property IdFornecedor() As Integer
        Property Comentario() As String 'Usar o Text.StringBuilder na implementação.
      End Interface

      Public Interface iKit
        Inherits Interfaces.Primitivas.iObjetoTabela
        ReadOnly Property idObj() As String
        Property Nome_key() As String
        Property ValorLocacao() As Double
        Property DescMax() As Double
        Property Instrucoes() As String 'Usar o Text.StringBuilder na implementação.
        Property Quantidade() As Integer
      End Interface

      Public Interface iKitItem
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property idKit_key() As Integer
        Property CodBarrasItem_key() As String
        Property Status() As Single
        Property TransferidoPara() As String
        Property Descricao() As String
        Property ValorLocacao() As Double
      End Interface

      Public Interface iNotaFiscal
        Inherits Interfaces.Primitivas.iObjetoTabela
        'Property idNotaFiscal() As Integer
        Property numeroFiscal_key() As Integer
        Property idOrdemServico() As Integer
        Property cancelada() As Boolean
        Property beneficiario() As String
        Property endereco() As String
        Property municipio() As String
        Property UF() As String
        Property CNPJ() As String
        Property IE() As String
        Property IM() As String
        Property formaPagamento() As String
        Property fone() As String
        Property dataFiscal() As Date
      End Interface

      Public Interface iNotaFiscalItem
        Inherits Interfaces.Primitivas.iObjetoTabela
        'Property idNotaFiscalItem() As Integer
        Property idNotaFiscal_key() As Integer
        Property unidade() As String
        Property quantidade() As Integer
        Property descricao_key() As String
        Property valor() As Decimal
      End Interface

      Public Interface iOrcamento
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property Confirmado() As Short
        Property Descricao_Key() As String
        Property DataEmissao_Key() As Date
        Property DataValidade() As Date
        Property DataVencimento() As Date
        Property IdLocalidade() As Integer
        Property IdEntidade() As Integer
        Property FormaPagamento() As String
        Property PercDesconto() As Double
        Property Contato() As String
        Property IsModelo() As Short
        Property Valor() As Double
      End Interface

      Public Interface iOrcamentoDespAdicional
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdOrcamento_Key() As Integer
        Property Descricao_Key() As String
        Property Quantidade() As Integer
        Property ValorDespesa() As Double
      End Interface

      Public Interface iOrcamentoObjs
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdOrcamento_key() As Integer
        Property IdObj_key() As String
        Property DataLocacaoInicio_key() As Date
        Property DataLocacaoFinal() As Date
        Property Descricao() As String
        Property Quantidade() As Double
        Property ValorUnitario() As Double
        Property ValorLocacao() As Double
        Property CtrObj() As iCtrObj
      End Interface

      Public Interface iOrcamentoParcelas
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdOrcamento_key() As Integer
        Property NumeroParcela_key() As Integer
        Property DataVencimento() As Date
        Property ValorParcela() As Double
      End Interface

      Public Interface iOrdemServico
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdOrcamento() As Integer
        Property DataEmissao() As Date
        Property Descricao() As String
        Property IdUsuario() As Integer
        Property DataAlteracao() As Date
      End Interface

      Public Interface iOrdemServicoObjs
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdOS() As Integer
        Property IdObj() As String
        Property DataDevolucao() As Date
        Property Quantidade() As Integer
        Property CtrObj() As iCtrObj
      End Interface

      Public Interface iStatusReal
        Inherits Interfaces.Primitivas.iObjetoTabela

        Property IdItem() As Integer
        Property IdKit() As Integer
        Property IdObj() As String
        Property IdEntidade() As Integer
        Property CodBarras() As String
        Property DataDevolucao() As Date
        Property Status() As Single
      End Interface

      Public Interface iStatusHistorico
        Inherits Interfaces.Primitivas.iObjetoTabela

        Property IdItem() As Integer
        Property IdKit() As Integer
        Property IdObj() As String
        Property IdEntidade() As Integer
        Property Dia() As Date
        Property CodBarras() As String
        Property Status() As Single
      End Interface

      Public Interface iStatusPresumido
        Inherits Interfaces.Primitivas.iObjetoTabela

        Property IdItem() As Integer
        Property IdKit() As Integer
        Property idOrcObj() As Integer
        Property IdObj_Key() As String
        Property IdOrcamento_Key() As Integer
        Property Dia_Key() As Date
        Property Quantidade() As Integer
        Property Status() As Single
      End Interface

      Public Interface iLinkItemNo
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property idNo() As Integer
        Property idItem() As Integer
      End Interface

      Public Interface iHistoricoItem
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property idFix() As Integer
        Property idItem_key() As Integer
        Property idSituacao_key() As Integer
        Property Data_key() As DateTime
        Property Hora() As DateTime
        Property idUsuario() As Integer
        Property IP() As String
      End Interface

      Public Interface iContasPagarReceber
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property TipoConta() As Integer
        Property Descricao() As String
        Property Vencimento() As Date
        Property ValorTotal() As Double
        Property Parcelas() As Integer
        Property AnotacoesAdicionais() As String
      End Interface

      Public Interface iContasPagarReceber_Parcelas
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property idConta() As Integer
        Property Data() As Date
        Property Valor() As Double
      End Interface

      Public Interface iCtrStatus
        Sub SalvarPresumido()
        Sub SalvarReal()
      End Interface

      Public Interface iCtrObj
        ReadOnly Property IdObj() As String
        Sub SalvarPresumido(ByVal noCommit As Boolean, ByVal idorcamento As Integer, ByVal idorcobj As Integer)
      End Interface

      Public Interface iCustomROrc
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdOrcamento() As Integer
        Property PathImgCab() As String
        Property TxtApresentacao() As String
      End Interface

      Public Interface iCustomROrc_Grupos
        Inherits Interfaces.Primitivas.iObjetoTabela
        Property IdCustomRO() As Integer
        Property Titulo() As String
        Property SubTitulo() As String
      End Interface

      Public Interface iCustomROrc_ItensGrupo
        Inherits System.IDisposable
        Property IdCustomRO_Grupos() As Integer
        Property IdObj() As String
        Property Quantidade() As Integer
        Property Descricao() As String
        Property Marca() As String
        Property Modelo() As String
        Sub Salvar(ByVal nocommit As Boolean)
        Sub Salvar()
      End Interface

    End Namespace
  End Namespace
End Namespace

File: src/NBdbm/CS/Classes/Interfaces/iEVT.cs
using System.Data;
using System.Collections;
using Microsoft.VisualBasic;
using System.Diagnostics;
using System;

namespace NBdbm.Interfaces.iEVT
{
    public interface iCadastroFuncionarios : NBdbm.Interfaces.iCTR.iCadastroUsuario
    {
    }

    public interface iCadastroLocalidades
    {
        //Inherits NBdbm.Interfaces.iCTR.iCadastroEntidade
        void Salvar(bool NoCommit);
        void Salvar();
        void Excluir(bool NoCommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iLocalidades Localidade
        {
            get;
            set;
        }
    }

    public interface iCadastroItens : System.IDisposable
    {
        void Salvar(bool NoCommit);
        void Salvar();
        void Excluir(bool NoCommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iItem Item
        {
            get;
            set;
        }
        string xmPath_LinkItemNo
        {
            get;
            set;
        }
    }

    public interface iCadatroKit : System.IDisposable
    {
        //Inherits Interfaces.Primitivas.allClass - > Não herda, porque ela só aglutina!
        void Salvar(bool NoCommit);
        void Salvar();
        void Excluir(bool NoCommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iKit Kit
        {
            get;
            set;
        }
        //Property xmPath_LinkEntNo() As String 'Se houver nó para os kits
        Interfaces.iEVT.Primitivas.iKitItem KitItem
        {
            get;
            set;
        }
    }

    public interface iCadatroOrcamento : System.IDisposable
    {
        //Inherits Interfaces.Primitivas.allClass - > Não herda, porque ela só aglutina!
        void Salvar(bool NoCommit);
        void Salvar();
        void Excluir(bool NoCommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iOrcamento Orc
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iOrcamentoObjs OrcItem
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iOrcamentoObjs OrcKit
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iOrcamentoDespAdicional OrcDespAdic
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iOrcamentoParcelas OrcParcela
        {
            get;
            set;
        }
    }

    public interface iCadastroCustomROrc : System.IDisposable
    {
        void Salvar(bool nocommit);
        void Salvar();
        void Excluir(bool nocommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iCustomROrc CustomROrc
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iCustomROrc_Grupos CustomROrc_Grupos
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iCustomROrc_ItensGrupo CustomROrc_ItemGrupo
        {
            get;
            set;
        }
        System.Collections.Hashtable ColCustomROrc_Grupos
        {
            get;
        }
        System.Collections.Hashtable ColCustomROrc_ItensGrupo
        {
            get;
        }
    }

    public interface iCadatroOrdemServico : System.IDisposable
    {
        void Salvar(bool NoCommit);
        void Salvar();
        void Excluir(bool NoCommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iOrdemServico OS
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iOrdemServicoObjs OSObjs
        {
            get;
            set;
        }
    }

    public interface iCadastroContasPagarReceber : IDisposable
    {
        void Salvar(bool NoCommit);
        void Salvar();
        void Excluir(bool NoCommit);
        void Excluir();
        Interfaces.iEVT.Primitivas.iContasPagarReceber Conta
        {
            get;
            set;
        }
        Interfaces.iEVT.Primitivas.iContasPagarReceber_Parcelas Parcela
        {
            get;
            set;
        }
        NBdbm.Fachadas.NbCollection Parcelas
        {
            get;
        }
    }


    namespace Primitivas
    {

        public interface iQualidade : Interfaces.Primitivas.iObjetoTabela
        {
            string qualidade_key
            {
                get;
                set;
            }
        }

        public interface iSituacao : Interfaces.Primitivas.iObjetoTabela
        {
            string situacao_key
            {
                get;
                set;
            }
        }

        public interface iLocalidades : Interfaces.Primitivas.iObjetoTabela
        {
            int idEntidade
            {
                get;
                set;
            }
            string Nome_key
            {
                get;
                set;
            }
            string comoChegar
            {
                get;
                set;
            }
        }

        public interface iDeposito : Interfaces.Primitivas.iObjetoTabela
        {
            string Nome_Key
            {
                get;
                set;
            }
            string Endereco
            {
                get;
                set;
            }
        }

        public interface iItem : Interfaces.Primitivas.iObjetoTabela
        {

            string IdObj
            {
                get;
            }
            string Descricao_key
            {
                get;
                set;
            }
            int Quantidade
            {
                get;
                set;
            }
            double ValorLocacao
            {
                get;
                set;
            }
            decimal DescMax
            {
                get;
                set;
            }
            int VidaUtil
            {
                get;
                set;
            }
            string NumeroSerie
            {
                get;
                set;
            }
            int IdQualidade
            {
                get;
                set;
            }
            int IdDeposito
            {
                get;
                set;
            }
            int IdFornecedor
            {
                get;
                set;
            }
            string Comentario
            { //Usar o Text.StringBuilder na implementação.
                get;
                set;
            }
        }

        public interface iKit : Interfaces.Primitivas.iObjetoTabela
        {
            string idObj
            {
                get;
            }
            string Nome_key
            {
                get;
                set;
            }
            double ValorLocacao
            {
                get;
                set;
            }
            double DescMax
            {
                get;
                set;
            }
            string Instrucoes
            { //Usar o Text.StringBuilder na implementação.
                get;
                set;
            }
            int Quantidade
            {
                get;
                set;
            }
        }

        public interface iKitItem : Interfaces.Primitivas.iObjetoTabela
        {
            int idKit_key
            {
                get;
                set;
            }
            string CodBarrasItem_key
            {
                get;
                set;
            }
            float Status
            {
                get;
                set;
            }
            string TransferidoPara
            {
                get;
                set;
            }
            string Descricao
            {
                get;
                set;
            }
            double ValorLocacao
            {
                get;
                set;
            }
        }

        public interface iNotaFiscal : Interfaces.Primitivas.iObjetoTabela
        {
            //Property idNotaFiscal() As Integer
            int numeroFiscal_key
            {
                get;
                set;
            }
            int idOrdemServico
            {
                get;
                set;
            }
            bool cancelada
            {
                get;
                set;
            }
            string beneficiario
            {
                get;
                set;
            }
            string endereco
            {
                get;
                set;
            }
            string municipio
            {
                get;
                set;
            }
            string UF
            {
                get;
                set;
            }
            string CNPJ
            {
                get;
                set;
            }
            string IE
            {
                get;
                set;
            }
            string IM
            {
                get;
                set;
            }
            string formaPagamento
            {
                get;
                set;
            }
            string fone
            {
                get;
                set;
            }
            DateTime dataFiscal
            {
                get;
                set;
            }
        }

        public interface iNotaFiscalItem : Interfaces.Primitivas.iObjetoTabela
        {
            //Property idNotaFiscalItem() As Integer
            int idNotaFiscal_key
            {
                get;
                set;
            }
            string unidade
            {
                get;
                set;
            }
            int quantidade
            {
                get;
                set;
            }
            string descricao_key
            {
                get;
                set;
            }
            decimal valor
            {
                get;
                set;
            }
        }

        public interface iOrcamento : Interfaces.Primitivas.iObjetoTabela
        {
            short Confirmado
            {
                get;
                set;
            }
            string Descricao_Key
            {
                get;
                set;
            }
            DateTime DataEmissao_Key
            {
                get;
                set;
            }
            DateTime DataValidade
            {
                get;
                set;
            }
            DateTime DataVencimento
            {
                get;
                set;
            }
            int IdLocalidade
            {
                get;
                set;
            }
            int IdEntidade
            {
                get;
                set;
            }
            string FormaPagamento
            {
                get;
                set;
            }
            double PercDesconto
            {
                get;
                set;
            }
            string Contato
            {
                get;
                set;
            }
            short IsModelo
            {
                get;
                set;
            }
            double Valor
            {
                get;
                set;
            }
        }

        public interface iOrcamentoDespAdicional : Interfaces.Primitivas.iObjetoTabela
        {
            int IdOrcamento_Key
            {
                get;
                set;
            }
            string Descricao_Key
            {
                get;
                set;
            }
            int Quantidade
            {
                get;
                set;
            }
            double ValorDespesa
            {
                get;
                set;
            }
        }

        public interface iOrcamentoObjs : Interfaces.Primitivas.iObjetoTabela
        {
            int IdOrcamento_key
            {
                get;
                set;
            }
            string IdObj_key
            {
                get;
                set;
            }
            DateTime DataLocacaoInicio_key
            {
                get;
                set;
            }
            DateTime DataLocacaoFinal
            {
                get;
                set;
            }
            string Descricao
            {
                get;
                set;
            }
            double Quantidade
            {
                get;
                set;
            }
            double ValorUnitario
            {
                get;
                set;
            }
            double ValorLocacao
            {
                get;
                set;
            }
            iCtrObj CtrObj
            {
                get;
                set;
            }
        }

        public interface iOrcamentoParcelas : Interfaces.Primitivas.iObjetoTabela
        {
            int IdOrcamento_key
            {
                get;
                set;
            }
            int NumeroParcela_key
            {
                get;
                set;
            }
            DateTime DataVencimento
            {
                get;
                set;
            }
            double ValorParcela
            {
                get;
                set;
            }
        }

        public interface iOrdemServico : Interfaces.Primitivas.iObjetoTabela
        {
            int IdOrcamento
            {
                get;
                set;
            }
            DateTime DataEmissao
            {
                get;
                set;
            }
            string Descricao
            {
                get;
                set;
            }
            int IdUsuario
            {
                get;
                set;
            }
            DateTime DataAlteracao
            {
                get;
                set;
            }
        }

        public interface iOrdemServicoObjs : Interfaces.Primitivas.iObjetoTabela
        {
            int IdOS
            {
                get;
                set;
            }
            string IdObj
            {
                get;
                set;
            }
            DateTime DataDevolucao
            {
                get;
                set;
            }
            int Quantidade
            {
                get;
                set;
            }
            iCtrObj CtrObj
            {
                get;
                set;
            }
        }

        public interface iStatusReal : Interfaces.Primitivas.iObjetoTabela
        {

            int IdItem
            {
                get;
                set;
            }
            int IdKit
            {
                get;
                set;
            }
            string IdObj
            {
                get;
                set;
            }
            int IdEntidade
            {
                get;
                set;
            }
            string CodBarras
            {
                get;
                set;
            }
            DateTime DataDevolucao
            {
                get;
                set;
            }
            float Status
            {
                get;
                set;
            }
        }

        public interface iStatusHistorico : Interfaces.Primitivas.iObjetoTabela
        {

            int IdItem
            {
                get;
                set;
            }
            int IdKit
            {
                get;
                set;
            }
            string IdObj
            {
                get;
                set;
            }
            int IdEntidade
            {
                get;
                set;
            }
            DateTime Dia
            {
                get;
                set;
            }
            string CodBarras
            {
                get;
                set;
            }
            float Status
            {
                get;
                set;
            }
        }

        public interface iStatusPresumido : Interfaces.Primitivas.iObjetoTabela
        {

            int IdItem
            {
                get;
                set;
            }
            int IdKit
            {
                get;
                set;
            }
            int idOrcObj
            {
                get;
                set;
            }
            string IdObj_Key
            {
                get;
                set;
            }
            int IdOrcamento_Key
            {
                get;
                set;
            }
            DateTime Dia_Key
            {
                get;
                set;
            }
            int Quantidade
            {
                get;
                set;
            }
            float Status
            {
                get;
                set;
            }
        }

        public interface iLinkItemNo : Interfaces.Primitivas.iObjetoTabela
        {
            int idNo
            {
                get;
                set;
            }
            int idItem
            {
                get;
                set;
            }
        }

        public interface iHistoricoItem : Interfaces.Primitivas.iObjetoTabela
        {
            int idFix
            {
                get;
                set;
            }
            int idItem_key
            {
                get;
                set;
            }
            int idSituacao_key
            {
                get;
                set;
            }
            DateTime Data_key
            {
                get;
                set;
            }
            DateTime Hora
            {
                get;
                set;
            }
            int idUsuario
            {
                get;
                set;
            }
            string IP
            {
                get;
                set;
            }
        }

        public interface iContasPagarReceber : Interfaces.Primitivas.iObjetoTabela
        {
            int TipoConta
            {
                get;
                set;
            }
            string Descricao
            {
                get;
                set;
            }
            DateTime Vencimento
            {
                get;
                set;
            }
            double ValorTotal
            {
                get;
                set;
            }
            int Parcelas
            {
                get;
                set;
            }
            string AnotacoesAdicionais
            {
                get;
                set;
            }
        }

        public interface iContasPagarReceber_Parcelas : Interfaces.Primitivas.iObjetoTabela
        {
            int idConta
            {
                get;
                set;
            }
            DateTime Data
            {
                get;
                set;
            }
            double Valor
            {
                get;
                set;
            }
        }

        public interface iCtrStatus
        {
            void SalvarPresumido();
            void SalvarReal();
        }

        public interface iCtrObj
        {
            string IdObj
            {
                get;
            }
            void SalvarPresumido(bool noCommit, int idorcamento, int idorcobj);
        }

        public interface iCustomROrc : Interfaces.Primitivas.iObjetoTabela
        {
            int IdOrcamento
            {
                get;
                set;
            }
            string PathImgCab
            {
                get;
                set;
            }
            string TxtApresentacao
            {
                get;
                set;
            }
        }

        public interface iCustomROrc_Grupos : Interfaces.Primitivas.iObjetoTabela
        {
            int IdCustomRO
            {
                get;
                set;
            }
            string Titulo
            {
                get;
                set;
            }
            string SubTitulo
            {
                get;
                set;
            }
        }

        public interface iCustomROrc_ItensGrupo : System.IDisposable
        {
            int IdCustomRO_Grupos
            {
                get;
                set;
            }
            string IdObj
            {
                get;
                set;
            }
            int Quantidade
            {
                get;
                set;
            }
            string Descricao
            {
                get;
                set;
            }
            string Marca
            {
                get;
                set;
            }
            string Modelo
            {
                get;
                set;
            }
            void Salvar(bool nocommit);
            void Salvar();
        }

    }
}
